src/hub/Hub.sol

```diff
@@ -62,6 +59,15 @@ contract Hub is Multicall, Auth, Recoverable, IHub, IHubGatewayHandler {
         transientValuation = transientValuation_;
     }
 
+    /// @inheritdoc IHub
+    modifier unlocked(PoolId poolId) {
+        require(hubRegistry.isAdmin(poolId, msg.sender), IHub.NotAuthorizedAdmin());
+
+        accounting.unlock(poolId);
+        _;
+        accounting.lock();
```

Comment:
I propose to extract these lines to a new `modifier unlocking()` and add only to the methods that really need to touch the accounting

Reply:
Updated!